### PR TITLE
[CIEDEV-3991][Bundle Authoring]: Create static container image with terraform binaries

### PR DESCRIPTION
**Pull request by anjanikshree12**
## This PR includes
- Docker container to serve terraform binary `terraform_1.4.7_linux_amd64.zip` as zip using nginx as a static file server.
- Updated workflows to build and push the image to Github Container Registry
- Updated github actions versions to fix github workflow warnings.
- Updated github actions for release action & tag manager.
## Testing Details
- Created a test build: [Build](https://github.com/terrariumcloud/terrarium/actions/runs/9680723989) (Uploads the images to GHCR & creates a release)
- Pulled the image from GHCR:
image
- Ran the image locally and tested for working of the container
Screenshot 2024-06-26 194621

### CODE REVIEW

File: .github/workflows/build-push-release.yml [hunk 1]
```diff
@@ -8,21 +8,47 @@ on:
   workflow_dispatch:
 
 jobs:
+  build-push-nginx-static-server:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
```
Comment (adcharre) [inline .github/workflows/build-push-release.yml:16]:
```suggestion
      uses: actions/checkout@v4
```

File: .github/workflows/build-push-release.yml [hunk 2]
```diff
@@ -8,21 +8,47 @@ on:
...
+        tags: ghcr.io/terrariumcloud/nginx-static-server:latest
+
   build-push:
     runs-on: ubuntu-latest
+    needs: [build-push-nginx-static-server]
+
     steps:
     - name: Checkout Code
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
```
Comment (adcharre) [inline .github/workflows/build-push-release.yml:41]:
```suggestion
      uses: actions/checkout@v4
```

File: .github/workflows/build-push-release.yml [hunk 3]
```diff
@@ -8,21 +8,47 @@ on:
...
 
     - name: Login to Github Container Registry
-      uses: docker/login-action@v1 
+      uses: docker/login-action@v2
```
Comment (adcharre) [inline .github/workflows/build-push-release.yml:44]:
```suggestion
      uses: docker/login-action@v3
```

File: .github/workflows/build-push-release.yml [hunk 4]
```diff
@@ -8,21 +8,47 @@ on:
...
 jobs:
+  build-push-nginx-static-server:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
+
+    - name: Login to Github Container Registry
+      uses: docker/login-action@v2
```
Comment (adcharre) [inline .github/workflows/build-push-release.yml:19]:
```suggestion
      uses: docker/login-action@v3
```

File: .github/workflows/build-push-release.yml [hunk 5]
```diff
@@ -8,21 +8,47 @@ on:
...
-      uses: docker/login-action@v1 
+      uses: docker/login-action@v2
       with:
         registry: ghcr.io
         username: ${{ github.actor }}
         password: ${{ secrets.GITHUB_TOKEN }}
     
     - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v1
+      uses: docker/setup-buildx-action@v2
```
Comment (adcharre) [inline .github/workflows/build-push-release.yml:51]:
```suggestion
      uses: docker/setup-buildx-action@v3
```

File: .github/workflows/build-push-release.yml [hunk 6]
```diff
@@ -8,21 +8,47 @@ on:
...
+
+    - name: Login to Github Container Registry
+      uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
```
Comment (adcharre) [inline .github/workflows/build-push-release.yml:26]:
```suggestion
      uses: docker/setup-buildx-action@v3
```

File: .github/workflows/build-push-release.yml [hunk 7]
```diff
@@ -8,21 +8,47 @@ on:
...
+    
+    - name: Build and push Terraform static server image
+      uses: docker/build-push-action@v3
```
Comment (adcharre) [inline .github/workflows/build-push-release.yml:29]:
```suggestion
      uses: docker/build-push-action@v6
```

File: .github/workflows/build-push-release.yml [hunk 8]
```diff
@@ -31,7 +57,7 @@ jobs:
         github_token: ${{ secrets.GITHUB_TOKEN }}
   
     - name: Build and push Docker image
-      uses: docker/build-push-action@v2
+      uses: docker/build-push-action@v3
```
Comment (adcharre) [inline .github/workflows/build-push-release.yml:60]:
```suggestion
      uses: docker/build-push-action@v6
```

File: containers/nginx-static-server/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,12 @@
+FROM nginx:latest
+
+# Set working directory to the Nginx document root
+WORKDIR /usr/share/nginx/html
+
+# Download the Terraform binary zip file, clean up and reduce the size of the Docker image by removing unnecessary files after installing packages.
+RUN apt-get update && apt-get install -y curl && \
```
Comment (adcharre) [inline containers/nginx-static-server/Dockerfile:7]: Please add this to a directory with the version name, ie download to 
`1.4.7/terraform_1.4.7_linux_amd64.zip`

**Review by adcharre — COMMENTED** [review]
Update all github actions to their latest versions and place the downloaded terraform binary in a version directory (so we can support more than 1 version in the same structure as the hashicorp site).

File: containers/nginx-static-server/Dockerfile (hunk 9 above)
Comment (anjanikshree12) [reply containers/nginx-static-server/Dockerfile:7]: yup, Updated.
Infact updated it to a directory something like `terraform/1.4.7/terraform_1.4.7_linux_amd64.zip`

**Review by adcharre — APPROVED** [review]

**Pull request merged**